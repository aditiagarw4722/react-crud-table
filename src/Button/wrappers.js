import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-button');

export const Btn = block('button');

export default { Btn };
