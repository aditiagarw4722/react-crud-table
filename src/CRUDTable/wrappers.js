import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table');

export const Table = block('table');

Table.displayName = 'Table';

Table.Row = element('tr', 'row');
Table.Row.displayName = 'Row';

Table.HeaderCell = element('th', 'header-cell');
Table.HeaderCell.displayName = 'HeaderCell';

Table.Header = element('thead', 'header');
Table.Header.displayName = 'Header';

Table.Body = element('tbody', 'body');
Table.Body.displayName = 'Body';

Table.Cell = element('td', 'cell');
Table.Cell.displayName = 'Cell';

Table.CellLabel = element('div', 'cell-label');
Table.CellLabel.displayName = 'CellLabel';

Table.Caption = element('div', 'caption');
Table.Caption.displayName = 'Caption';

export default { Table };
