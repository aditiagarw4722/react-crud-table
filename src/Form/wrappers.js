import { Form as FormikForm } from 'formik';
import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-form');

export const Form = block(FormikForm);

Form.displayName = 'FormikForm';

Form.Label = element('label', 'label');
Form.Label.displayName = 'Label';

Form.Message = element('div', 'message');
Form.Message.displayName = 'Message';

Form.ErrorMessage = element('div', 'error-message');
Form.ErrorMessage.displayName = 'ErrorMessage';

Form.FieldError = element('div', 'field-error');
Form.FieldError.displayName = 'FieldError';

Form.FieldContainer = element('div', 'field-container');
Form.FieldContainer.displayName = 'FieldContainer';

export default {
  Form,
};
