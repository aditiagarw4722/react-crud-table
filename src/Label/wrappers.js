import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-label');

const Label = block('div');
Label.displayName = 'Label';

export default Label;
