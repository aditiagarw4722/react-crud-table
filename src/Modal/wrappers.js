import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-wrapper');

export const Container = block('div');
Container.BG = element('div', 'background');
Container.Modal = element('div', 'modal');
Container.Title = element('h3', 'title');

export default {
  Container,
};
