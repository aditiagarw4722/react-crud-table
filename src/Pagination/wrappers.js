import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-pagination');

export const Pagination = block('div');

Pagination.Link = element('button', 'link');
Pagination.Prev = element('button', 'previous');
Pagination.Next = element('button', 'next');

export default { Pagination };
