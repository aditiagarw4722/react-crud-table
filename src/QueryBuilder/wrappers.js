import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-query-builder');

export const Container = block('div');

Container.displayName = 'Container';

export const RuleBuilder = element('div', 'search-rule-builder');

RuleBuilder.displayName = 'RuleBuilder';

export default { Container, RuleBuilder };
