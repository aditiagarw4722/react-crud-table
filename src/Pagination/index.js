import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Pagination as Container } from './wrappers';

class Pagination extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activePage: props.defaultActivePage,
    };
    this.handleNextClick = this.handleNextClick.bind(this);
    this.handlePreviousClick = this.handlePreviousClick.bind(this);
    this.handleLinkClick = this.handleLinkClick.bind(this);
  }

  handlePreviousClick() {
    const { activePage } = this.state;
    if (activePage > 1) {
      this.update(activePage - 1);
    }
  }

  handleNextClick() {
    const { activePage } = this.state;
    const numberOfPages = this.calculateNumberOfPages();
    if (activePage < numberOfPages) {
      this.update(activePage + 1);
    }
  }

  handleLinkClick(evt) {
    this.update(+evt.target.textContent.trim());
  }

  update(activePage) {
    const { totalOfItems, itemsPerPage, onChange } = this.props;
    this.setState({ activePage });
    onChange({
      activePage,
      totalOfItems,
      itemsPerPage,
    });
  }

  calculateNumberOfPages() {
    const { totalOfItems, itemsPerPage } = this.props;
    return Math.ceil(totalOfItems / itemsPerPage);
  }

  render() {
    const { activePage } = this.state;
    const numberOfPages = this.calculateNumberOfPages();
    const numbers = [...Array(numberOfPages).keys()];
    return (
      <Container>
        <Container.Prev onClick={this.handlePreviousClick}>
          &laquo;
        </Container.Prev>
        {numbers.map((i) => (
          <Container.Link
            key={i}
            modifiers={i === activePage ? 'active' : 'inactive'}
            onClick={this.handleLinkClick}
          >
            {i}
          </Container.Link>
        ))}
        <Container.Next onClick={this.handleNextClick}>&raquo;</Container.Next>
      </Container>
    );
  }
}

Pagination.propTypes = {
  defaultActivePage: PropTypes.number,
  itemsPerPage: PropTypes.number,
  totalOfItems: PropTypes.number,
  onChange: PropTypes.func,
};

Pagination.defaultProps = {
  defaultActivePage: 1,
  totalOfItems: 0,
  itemsPerPage: 10,
  onChange: () => {},
};

export default Pagination;
