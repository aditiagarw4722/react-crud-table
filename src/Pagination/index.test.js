import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import Enzyme from 'enzyme';

import { Pagination as Container } from './wrappers';

import Pagination from '.';

const renderer = new ShallowRenderer();

describe('Pagination', () => {
  let onChange;

  beforeEach(() => {
    onChange = jest.fn();
  });

  it('renders as expected when in the first page', () => {
    const result = renderer.render(
      <Pagination
        activePage={1}
        totalOfItems={10}
        itemsPerPage={2}
        onPageChange={onChange}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('renders as expected when in the last page', () => {
    const result = renderer.render(
      <Pagination
        activePage={5}
        totalOfItems={10}
        itemsPerPage={2}
        onPageChange={onChange}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('renders as expected when in a page in the middle', () => {
    const result = renderer.render(
      <Pagination
        activePage={3}
        totalOfItems={10}
        itemsPerPage={2}
        onPageChange={onChange}
      />
    );
    expect(result).toMatchSnapshot();
  });

  const Foo = () => <div />;

  it('notifies a page change when clicking a numbered link', () => {
    const result = create(
      <Pagination
        activePage={3}
        totalOfItems={10}
        itemsPerPage={2}
        onPageChange={onChange}
      />
    );

    result.root.findByType(Container.Prev).props.onClick();
    result.root.findByType(Container.Next).props.onClick();
    result.root.findAllByType(Container.Link).forEach((item) => {
      item.props.onClick();
    });

    expect(onChange).toHaveBeenCalledTimes(7);
    expect(onChange.mock.calls).toEqual([[2], [4], [1], [2], [3], [4], [5]]);
  });
});
