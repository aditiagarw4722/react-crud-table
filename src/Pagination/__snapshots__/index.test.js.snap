// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders as expected when in a page in the middle 1`] = `
<Pagination>
  <PaginationPrev
    onClick={[Function]}
  >
    «
  </PaginationPrev>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    1
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    2
  </PaginationLink>
  <PaginationLink
    modifiers="active"
    onClick={[Function]}
  >
    3
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    4
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    5
  </PaginationLink>
  <PaginationNext
    onClick={[Function]}
  >
    »
  </PaginationNext>
</Pagination>
`;

exports[`Pagination renders as expected when in the first page 1`] = `
<Pagination>
  <PaginationLink
    modifiers="active"
    onClick={[Function]}
  >
    1
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    2
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    3
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    4
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    5
  </PaginationLink>
  <PaginationNext
    onClick={[Function]}
  >
    »
  </PaginationNext>
</Pagination>
`;

exports[`Pagination renders as expected when in the last page 1`] = `
<Pagination>
  <PaginationPrev
    onClick={[Function]}
  >
    «
  </PaginationPrev>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    1
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    2
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    3
  </PaginationLink>
  <PaginationLink
    modifiers="inactive"
    onClick={[Function]}
  >
    4
  </PaginationLink>
  <PaginationLink
    modifiers="active"
    onClick={[Function]}
  >
    5
  </PaginationLink>
</Pagination>
`;

exports[`Pagination renders as expected when there are no items 1`] = `
<Pagination>
  <PaginationLink
    modifiers="active"
    onClick={[Function]}
  >
    1
  </PaginationLink>
</Pagination>
`;
